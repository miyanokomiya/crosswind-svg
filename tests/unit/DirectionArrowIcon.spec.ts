import { mount } from "@vue/test-utils";
import Target from "../../src/components/DirectionArrowIcon.vue";

describe("DirectionArrowIcon", () => {
  it("to:u", () => {
    const wrapper = mount(Target, {
      propsData: { to: "u" }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it("to:r", () => {
    const wrapper = mount(Target, {
      propsData: { to: "r" }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it("to:l", () => {
    const wrapper = mount(Target, {
      propsData: { to: "l" }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
  it("to:d", () => {
    const wrapper = mount(Target, {
      propsData: { to: "b" }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
