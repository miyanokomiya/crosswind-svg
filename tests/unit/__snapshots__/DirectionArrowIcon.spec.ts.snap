// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DirectionArrowIcon to:d 1`] = `
<svg
  style="transform: rotate(0); fill: currentColor; stroke: currentColor;"
  viewBox="0 0 64 64"
  xmlns="http://www.w3.org/2000/svg"
  xmlns:xlink="http://www.w3.org/1999/xlink"
>
  <g
    fill="white"
  >
    <circle
      cx="32"
      cy="32"
      r="32"
    />
  </g>
   
  <g
    stroke=""
    stroke-linecap="round"
    stroke-width="4"
  >
    <path
      d="M 20 20 44 20"
      style="transform-origin: 50% 50%; transition: all 0.5s; transform: rotate(-45deg) scaleY(-1);"
    />
     
    <path
      d="M 20 32 44 32"
      style="transform-origin: 50% 50%; transition: all 0.5s; transform: rotate(-90deg);"
    />
     
    <path
      d="M 20 44 44 44"
      style="transform-origin: 50% 50%; transition: all 0.5s; transform: rotate(45deg) scaleX(-1);"
    />
  </g>
</svg>
`;

exports[`DirectionArrowIcon to:l 1`] = `
<svg
  style="transform: rotate(0); fill: currentColor; stroke: currentColor;"
  viewBox="0 0 64 64"
  xmlns="http://www.w3.org/2000/svg"
  xmlns:xlink="http://www.w3.org/1999/xlink"
>
  <g
    fill="white"
  >
    <circle
      cx="32"
      cy="32"
      r="32"
    />
  </g>
   
  <g
    stroke=""
    stroke-linecap="round"
    stroke-width="4"
  >
    <path
      d="M 20 20 44 20"
      style="transform-origin: 50% 50%; transition: all 0.5s; transform: rotate(-45deg);"
    />
     
    <path
      d="M 20 32 44 32"
      style="transform-origin: 50% 50%; transition: all 0.5s; transform: rotate(-180deg);"
    />
     
    <path
      d="M 20 44 44 44"
      style="transform-origin: 50% 50%; transition: all 0.5s; transform: rotate(45deg);"
    />
  </g>
</svg>
`;

exports[`DirectionArrowIcon to:r 1`] = `
<svg
  style="transform: rotate(0); fill: currentColor; stroke: currentColor;"
  viewBox="0 0 64 64"
  xmlns="http://www.w3.org/2000/svg"
  xmlns:xlink="http://www.w3.org/1999/xlink"
>
  <g
    fill="white"
  >
    <circle
      cx="32"
      cy="32"
      r="32"
    />
  </g>
   
  <g
    stroke=""
    stroke-linecap="round"
    stroke-width="4"
  >
    <path
      d="M 20 20 44 20"
      style="transform-origin: 50% 50%; transition: all 0.5s; transform: rotate(45deg);"
    />
     
    <path
      d="M 20 32 44 32"
      style="transform-origin: 50% 50%; transition: all 0.5s;"
    />
     
    <path
      d="M 20 44 44 44"
      style="transform-origin: 50% 50%; transition: all 0.5s; transform: rotate(-45deg);"
    />
  </g>
</svg>
`;

exports[`DirectionArrowIcon to:u 1`] = `
<svg
  style="transform: rotate(0); fill: currentColor; stroke: currentColor;"
  viewBox="0 0 64 64"
  xmlns="http://www.w3.org/2000/svg"
  xmlns:xlink="http://www.w3.org/1999/xlink"
>
  <g
    fill="white"
  >
    <circle
      cx="32"
      cy="32"
      r="32"
    />
  </g>
   
  <g
    stroke=""
    stroke-linecap="round"
    stroke-width="4"
  >
    <path
      d="M 20 20 44 20"
      style="transform-origin: 50% 50%; transition: all 0.5s; transform: rotate(45deg);"
    />
     
    <path
      d="M 20 32 44 32"
      style="transform-origin: 50% 50%; transition: all 0.5s;"
    />
     
    <path
      d="M 20 44 44 44"
      style="transform-origin: 50% 50%; transition: all 0.5s; transform: rotate(-45deg);"
    />
  </g>
</svg>
`;
